File: frontend/src/api/_shared/mapper/feed-source.mapper.ts
import { IFeedSource } from '@/model/FeedSource.model';
import { IApiFeedSource } from '../ApiFeedSource.type';

export const feedSourceMapper = {

    map(feedSource: IApiFeedSource): IFeedSource {
        return {
            reference: feedSource.reference,
            title: feedSource.title,
            description: feedSource.description,
            rssUrl: feedSource.rssUrl,
            websiteUrl: feedSource.websiteUrl,
            collection: feedSource.collection,
        };
    },

};

File: backend/MightyRSS/Models/Mappers/FeedSourceMapper.cs
using Data.Records;

namespace MightyRSS.Models.Mappers;

public static class FeedSourceMapper
{
    public static FeedSourceModel Map(FeedSourceRecord feedSource, UserFeedSourceRecord userFeedSource)
    {
        return new FeedSourceModel
        {
            Reference = feedSource.Reference,
            Title = userFeedSource.Title ?? feedSource.Title,
            Description = feedSource.Description,
            RssUrl = feedSource.RssUrl,
            WebsiteUrl = feedSource.WebsiteUrl
        };
    }
}